### PR TITLE
refactor(Inventory Purchasable)

### LINKED ISSUE #2251: (refactor) stock Select Inventory Purchasable
Add a filter to select products that have reached a purchase order point,

### PR DESCRIPTION
**Pull request by lomamech**
- Implement the way to filter inventory who requires a Purchase Order
- Update API for stock core filter the inventorys who Quantity is sup to
  zero when require_po is true
- Update E2E test for inventory stock add test for the Inventories who
  require purchase order and use property resetFilters for improve some
  test

closes #2251

### CODE REVIEW

File: server/controllers/stock/core.js [hunk 1]
```diff
@@ -450,6 +456,11 @@ function getInventoryQuantityAndConsumption(params) {
       if (status) {
         return rows.filter(row => row.status === status);
       }
+
+      if(require_po) {
```
Comment (jniles) [inline server/controllers/stock/core.js]: It looks like `status and `require_po` are exclusive.  Can both options be set?  If so, these shouldn't `return rows.filter()`, they should do something like:
```js
var filteredRows = rows;

if (status) {
  filteredRows = filteredRows.filter(row => row.status === status);
}

if (require_po) {
  filteredRows = filteredRows.filter(row => row.S_Q > 0);
}

return filteredRows;
```

File: client/src/modules/stock/inventories/modals/search.modal.html [hunk 2]
```diff
@@ -31,6 +31,15 @@
             <bh-clear on-clear="$ctrl.clear('inventory_uuid')"></bh-clear>
           </bh-inventory-select>
 
+          <div class="form-group checkbox">
```
Comment (jniles) [inline client/src/modules/stock/inventories/modals/search.modal.html]: There's no need for the `form-group` class here.

**Review by jniles — COMMENTED** [review]
@lomamech, this code is pretty good!

I'm curious - shouldn't almost all the inventory be needing purchase as soon as one stock item is consumed?  It looks like this basically filters on `quantity != S_MAX`.

If so, is it really that useful to have a clientside filter?  The server-side filter is useful because we can then populate a PO for a client, but I'm not so sure the checkbox is useful.  What do you think?

**Comment by lomamech** [comment]
The client-side filter was implemented just to allow inventory managers to evaluate the list of products requiring a purchase order, I will create another issue this issue will select through the Purchase Order module the inventories that requires a purchase order (Automatic Purchase Order)

**Review by jniles — APPROVED** [review]
LGTM!

bors r+

**Pull request merged**